sdk/storage/storage-queue/review/storage-queue.api.ts

```diff
@@ -0,0 +1,496 @@
+}
+
+// @public
+class AnonymousCredentialPolicy extends CredentialPolicy {
+  constructor(nextPolicy: RequestPolicy, options: RequestPolicyOptions);
+}
+
+// @public (undocumented)
+class BaseRequestPolicy implements RequestPolicy {
+  protected constructor(_nextPolicy: RequestPolicy, _options: RequestPolicyOptions);
+  // (undocumented)
+  readonly _nextPolicy: RequestPolicy;
+  // (undocumented)
+  readonly _options: RequestPolicyOptions;
+  log(logLevel: HttpPipelineLogLevel, message: string): void;
+  // (undocumented)
+  abstract sendRequest(webResource: WebResource): Promise<HttpOperationResponse>;
+  shouldLog(logLevel: HttpPipelineLogLevel): boolean;
+}
+
+// @public
+class Credential implements RequestPolicyFactory {
+  create(_nextPolicy: RequestPolicy, _options: RequestPolicyOptions): RequestPolicy;
+}
+
+// @public
+class CredentialPolicy extends BaseRequestPolicy {
+  sendRequest(request: WebResource): Promise<HttpOperationResponse>;
+  protected signRequest(request: WebResource): WebResource;
+}
+
+// @public
+export function deserializationPolicy(deserializationContentTypes?: DeserializationContentTypes): RequestPolicyFactory;
+
+// @public
+export function generateAccountSASQueryParameters(accountSASSignatureValues: AccountSASSignatureValues, sharedKeyCredential: SharedKeyCredential): SASQueryParameters;
+
+// @public
+export function generateQueueSASQueryParameters(queueSASSignatureValues: QueueSASSignatureValues, sharedKeyCredential: SharedKeyCredential): SASQueryParameters;
+
+// @public
+class HttpHeaders {
+  constructor(rawHeaders?: RawHttpHeaders);
+  clone(): HttpHeaders;
+  contains(headerName: string): boolean;
+  get(headerName: string): string | undefined;
+  headerNames(): string[];
+  headersArray(): HttpHeader[];
+  headerValues(): string[];
+  rawHeaders(): RawHttpHeaders;
+  remove(headerName: string): boolean;
+  set(headerName: string, headerValue: string | number): void;
+  toJson(): RawHttpHeaders;
+  toString(): string;
+}
+
+// @public
+interface HttpOperationResponse extends HttpResponse {
+  blobBody?: Promise<Blob>;
+  bodyAsText?: string | null;
+  parsedBody?: any;
+  parsedHeaders?: {
+    [key: string]: any;
+  }
+  readableStreamBody?: NodeJS.ReadableStream;
+}
+
+// @public
+enum HttpPipelineLogLevel {
+  ERROR = 1,
+  INFO = 3,
+  OFF = 0,
+  WARNING = 2
+}
+
+// @public
+interface IHttpClient extends RequestPolicy {
+}
+
+// @public
+interface IHttpPipelineLogger {
+  log(logLevel: HttpPipelineLogLevel, message: string): void;
+  minimumLogLevel: HttpPipelineLogLevel;
+}
+
+// @public
+interface IPRange {
+  end?: string;
+  start: string;
+}
+
+// @public
+class LoggingPolicyFactory implements RequestPolicyFactory {
+  constructor(loggingOptions?: RequestLogOptions);
+  create(nextPolicy: RequestPolicy, options: RequestPolicyOptions): LoggingPolicy;
+}
+
+// @public
+class MessageIdClient extends StorageClient {
```

Comment:
Let's investigate if we can flatten this ala .NET APIs.

Reply:
Will do

Reply:
#5547